### ISSUE TITLE
The "This field is required" error is too subtle, and messes up vertical spacing.

### ISSUE THREAD

**Issue by kfogel** [OpenTechStrategies/smoke-alarm-portal#22]
![screenshot from 2015-06-20 13 11 24](https://cloud.githubusercontent.com/assets/401111/8266141/cd135104-174e-11e5-872d-45718bb0934c.png)
(See attached screenshot.)  When you try to submit with a required field missing -- "Name" in this example -- the form doesn't submit, but instead puts up a "This field is required" error under that field.  However, the error message has two problems:

* It is too subtle.  The message the same white text in the same font & size as everything else.  No decoration appears around or next to the text field rectangle in question to highlight that it is where the missing information should go.
* Despite being too subtle in other ways, it messes up the vertical spacing on the form.  In the screenshot, you can see that "Name" is now farther away from "Address", vertically, then anything below is from anything else below.

(Note that although in this particular screenshot, the vertical spacing change partially compensates for the subtlety, by making the "Name" line stand out from the rest of the group, in real life there may be multiple required fields displaying the error, so that compensating effect would be lost anyway.)

The vertical spacing should stay unchanged, and instead the text box should get some kind of highlighting and/or a suitably alarming-looking icon next to it, and the "This field is required" message should appear next to that.  This seems to be the way most sites do it these days, and it's pretty easy for the user to understand.

**Comment by nttaylor** [comment]
The changes in 3ac3645 are good (I believe), though more needs to be done. I want to ask for others' thoughts: One of @kfogel's main points was that the appearance of error messages should not change the vertical spacing of the form elements. This behavior is less now, but still happens if the error message needs more horizontal space than is available to the right of the related field.

So do we use shorter messages? Maybe this could work for the name, phone, and zip code fields, but the length of the email field makes this difficult. Other possibilities?

I've also reduced the font size for error messages, though of course this makes them harder to read, and they are already hard to read just by being against the b&w background image. Should I put some kind of solid background color behind them like I did with the labels?

Btw one annoying thing was that after spending a long time struggling with why the bootstrap grid wasn't behaving the way I thought it should, I realized that a css rule imported from DCSOps.css was breaking it. I've commented out the link to that style sheet for now.
